server/app/repository/ApplicationRepository.java

```diff
@@ -295,6 +299,21 @@ public CompletionStage<Optional<ApplicationModel>> getApplication(long applicati
         executionContext.current());
   }
 
+  /** Updates the Applications table */
+  public void saveEligibilityDetermination(
+      ApplicationModel application, EligibilityDetermination eligibilityDetermination) {
+    try (Transaction transaction = database.beginTransaction()) {
```

Comment:
question (maybe blocking): I have not been following what Austin is doing with transactions too closely, but does this match the preferred pattern?

Reply:
In this case I don't think we need the `application.save()` on line 312, because iiuc the `.update()` call is also making the update query.

Then, given that we only have one database interaction in this method, we don't need wrap it in a transaction cause it's just one update query, not a "read, modify in memory, write" cycle that could be susceptible to a race condition

All that said, I don't think we need this method at all? imo it'd be nicer to save the eligibility determination when the application is converted from a draft to active, by passing the eligibility_determination into the call to `submitApplication()` on line 570 of `ApplicantService`